Source/ConstraintAttributes.swift

```diff
@@ -157,10 +157,10 @@ internal struct ConstraintAttributes: RawOptionSetType, BooleanType {
     }
 }
 internal func +=(inout left: ConstraintAttributes, right: ConstraintAttributes) {
-    left = (left | right)
+    left = left.intersect(right)
```

Comment:
yea I think this is wrong too -- it should be a union not a intersect @bendjones can you fix that?